server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/routes/MailRepositoriesRoutes.java

```diff
@@ -271,6 +283,100 @@ public void defineDeleteAll() {
         }, jsonTransformer);
     }
 
+    @PATCH
+    @Path("/{encodedUrl}/mails")
+    @ApiOperation(value = "Reprocessing all mails in that mailRepository")
+    @ApiImplicitParams({
+        @ApiImplicitParam(
+            required = false,
+            name = "queue",
+            paramType = "query parameter",
+            dataType = "String",
+            defaultValue = "spool",
+            example = "?offset=outgoing",
+            value = "Indicates in which queue the mails stored in the repository should be re-enqueued"),
+        @ApiImplicitParam(
+            required = false,
+            paramType = "query parameter",
+            name = "processor",
+            dataType = "String",
+            defaultValue = "absent",
+            example = "?processor=transport",
+            value = "If present, modifies the state property of the mail to allow their processing by a specific mail container processor.")
+    })
+    @ApiResponses(value = {
+        @ApiResponse(code = HttpStatus.CREATED_201, message = "Task is created", response = TaskIdDto.class),
+        @ApiResponse(code = HttpStatus.INTERNAL_SERVER_ERROR_500, message = "Internal server error - Something went bad on the server side."),
+        @ApiResponse(code = HttpStatus.BAD_REQUEST_400, message = "Bad request - unknown action")
+    })
+    public void defineReprocessAll() {
+        service.patch(MAIL_REPOSITORIES + "/:encodedUrl/mails", (request, response) -> {
+            String url = decodedRepositoryUrl(request);
+            enforceActionParameter(request);
```

Comment:
there is no action parameter document, right ?

Reply:
Right... Rush you know...